packages/web3-eth-accounts/src/account.ts

```diff
@@ -143,25 +143,19 @@ export const sign = (data: string, privateKey: Bytes): SignResult => {
 
 	const hash = hashMessage(data);
 
-	const [signature, recoverId] = signSync(hash.substring(2), privateKeyBuffer, {
-		// Makes signatures compatible with libsecp256k1
-		recovered: true,
-
-		// Returned signature should be in DER format ( non compact )
-		der: false,
-	});
-
-	const r = Buffer.from(signature.slice(0, 32));
-	const s = Buffer.from(signature.slice(32, 64));
-	const v = recoverId + 27;
+	const signature = secp256k1.sign(hash.substring(2), privateKeyBuffer);
+	const signatureBytes = signature.toCompactRawBytes();
+	const r = Buffer.from(signatureBytes.subarray(0, 32));
+	const s = Buffer.from(signatureBytes.subarray(32, 64));
```

Comment:
```suggestion
	const r = signature.r.toString(16);
	const s = signature.s.toString(16);
```
 we can directly use from sig obj, instead of converting from buffer and sub array.